services/blockstorage/sync/block_sync_gossip_client.go

```diff
@@ -0,0 +1,84 @@
+package sync
+
+import (
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+)
+
+type blockSyncGossipClient struct {
+	gossip    gossiptopics.BlockSync
+	storage   BlockSyncStorage
+	logger    log.BasicLogger
+	batchSize uint32
+	nodeKey   primitives.Ed25519PublicKey
+}
+
+func newBlockSyncGossipClient(
+	g gossiptopics.BlockSync,
+	s BlockSyncStorage,
+	l log.BasicLogger,
+	batchSize uint32,
+	pk primitives.Ed25519PublicKey) *blockSyncGossipClient {
```

Comment:
what you can do is instead of sending the node pk, you can send a `Func() primitives.Ed25519PublicKey`, and reference the function from the config; it's more future-proof

Reply:
makes sense, will fix everywhere